posix: xsi_realtime: portability.posix.xsi_realtime* report RAM overflow

**Describe the bug**
Tests 
- `tests/posix/xsi_realtime/portability.posix.xsi_realtime`
- `tests/posix/xsi_realtime/portability.posix.xsi_realtime.minimal`
- `tests/posix/xsi_realtime/portability.posix.xsi_realtime.newlib`
- `tests/posix/xsi_realtime/portability.posix.xsi_realtime.picolibc`

report RAM overflow.

Observed for `qemu_cortex_m3/ti_lm3s6965`

**To Reproduce**
Steps to reproduce the behavior:
1. go to your zephyr dir
2. call `$ZEPHYR_BASE/scripts/twister -T tests/posix/xsi_realtime -p qemu_cortex_m3/ti_lm3s6965 -v --inline-logs --overflow-as-errors`
3. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `bss' will not fit in region `RAM'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: region `RAM' overflowed by 39304 bytes
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain Zephyr SDK 0.17.0
 - Commit SHA or Version used: `v4.1.0-3474-gee940cb6dcba`

--- a/tests/posix/xsi_realtime/testcase.yaml
+++ b/tests/posix/xsi_realtime/testcase.yaml
@@ -9,6 +9,10 @@ common:
     - simulation
   min_flash: 64
   min_ram: 96
+  integration_platforms:
+    - qemu_x86
+    - qemu_cortex_a53
+    - qemu_riscv64
   platform_exclude:
     # linker_zephyr_pre0.cmd:140: syntax error (??)
     - qemu_xtensa/dc233c

--- a/tests/posix/xsi_realtime/testcase.yaml
+++ b/tests/posix/xsi_realtime/testcase.yaml
@@ -8,8 +8,7 @@ common:
     - arch
     - simulation
   min_flash: 64
-  min_ram: 32
-  timeout: 240
+  min_ram: 96
   platform_exclude:
     # linker_zephyr_pre0.cmd:140: syntax error (??)
     - qemu_xtensa/dc233c

